sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.aest.controller.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  GetDesignationAgreementDto,
+  GetDesignationAgreementsDto,
+  PendingDesignationDto,
+  UpdateDesignationDto,
+} from "./models/designation-agreement.model";
+import { DesignationAgreementServiceController } from "./designation-agreement.service.controller";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("designation-agreement")
+export class DesignationAgreementAESTController {
+  constructor(
+    private readonly designationAgreementServiceController: DesignationAgreementServiceController,
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Retrieve the designation agreement information and
+   * the associated locations approvals.
+   * @param designationId
+   * @returns designation agreement information.
+   */
+  @Get(":designationId")
+  async getDesignationAgreement(
+    @Param("designationId") designationId: number,
+  ): Promise<GetDesignationAgreementDto> {
+    return this.designationAgreementServiceController.getDesignationAgreement(
+      designationId,
+    );
+  }
+
+  /**
+   * Get the list of all the designations that belongs to
+   * the institution.
+   * @param institutionId
+   * @returns the list of all the designations that
+   * belongs to the institution.
+   */
+  @Get("institution/:institutionId")
+  async getDesignationAgreements(
+    @Param("institutionId") institutionId: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    return this.designationAgreementServiceController.getDesignationAgreements(
+      institutionId,
+    );
+  }
+
+  /**
+   * API to retrieve all pending designations.
+   * @param designationStatus
+   * @param searchCriteria to search designation.
+   * @returns Pending designations.
+   */
+  @Get("status/:designationStatus")
+  async getDesignationAgreementByStatus(
+    @Param("designationStatus") designationStatus: DesignationAgreementStatus,
+    @Query(PaginationParams.SearchCriteria) searchCriteria: string,
+  ): Promise<PendingDesignationDto[]> {
+    if (
+      !Object.values(DesignationAgreementStatus).includes(designationStatus)
+    ) {
+      throw new NotFoundException("Invalid designation agreement status.");
+    }
+    const pendingDesignations =
+      await this.designationAgreementService.getDesignationAgreementsByStatus(
+        designationStatus,
+        searchCriteria,
+      );
+    return pendingDesignations.map(
+      (pendingDesignation) =>
+        ({
+          designationId: pendingDesignation.id,
+          designationStatus: pendingDesignation.designationStatus,
+          submittedDate: pendingDesignation.submittedDate,
+          startDate: getISODateOnlyString(pendingDesignation.startDate),
+          endDate: getISODateOnlyString(pendingDesignation.endDate),
+          institutionName: pendingDesignation.institution.legalOperatingName,
```

Comment:
Can we call it what it really represents in institution entity _legalOperatingName_ to avoid confusion between _legalOperatingName_ and _operatingName_.

Reply:
DTO property renamed.